porter/main.py

```diff
@@ -73,6 +72,17 @@ class RetrievalOutcome(NamedTuple):
         cfrags: Dict
         errors: Dict
 
+    class CBDDecryptionOutcome(NamedTuple):
+        """
+        Simple object that stores the results and errors of CBD decryption operations across
+        one or more Ursulas.
+        """
+
+        encrypted_decryption_responses: Dict[
+            ChecksumAddress, EncryptedThresholdDecryptionResponse
+        ]
+        errors: Dict[ChecksumAddress, str]
```

Comment:
Just a question: If the decryption request fails, it must be because one of the participants failed. In that case, what do we do on the client side? Do we retry once and then rest our case or is there any other action we could take based on `errors` contents? At this point, the ritual is complete and validated.

Reply:
Good question. This is a possible improvement to our error handling. I think we've chatted previously (perhaps there is an issue somewhere) about possibly having integer error codes or something like that as part of the error information returned, then the errors can be contextualized appropriately eg. blip on connectivity vs something more problematic (eg. wrong request encrypting key used etc.). A blip in connectivity would signal a retry would possibly be successful.

That is something we can perhaps build upon.